### PR TITLE
[Enhancer] create Content Repository Enhancer

### PR DESCRIPTION
**Pull request by andrey1s**
| Q             | A
| ------------- | ---
| Bug fix?      | [no]
| New feature?  | [yes]
| BC breaks?    | [no]
| Deprecations? | [no]
| Tests pass?   | [yes]
| Fixed tickets | ~
| License       | MIT
| Doc PR        | [#745](https://github.com/symfony-cmf/symfony-cmf-docs/pull/745)

### CODE REVIEW

File: Enhancer/ContentRepositoryEnhancer.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    private $target;
+
+    /**
+     * @var string
+     */
+    private $source;
+
+    /**
+     * ContentRepositoryEnhancer constructor.
```
Comment (dbu) [inline Enhancer/ContentRepositoryEnhancer.php]: lets remove this line, it does not say anything.

File: Enhancer/ContentRepositoryEnhancer.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     *
+     * @param ContentRepositoryInterface $contentRepository
+     * @param string                     $target
+     * @param string                     $source
```
Comment (dbu) [inline Enhancer/ContentRepositoryEnhancer.php]: please add a short description that these are the field names and that the source is used as id on the content repository and target is the field where its written to

File: Enhancer/ContentRepositoryEnhancer.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    public function __construct(
+        ContentRepositoryInterface $contentRepository,
+        $target = RouteObjectInterface::CONTENT_OBJECT,
+        $source = RouteObjectInterface::CONTENT_ID)
+    {
```
Comment (dbu) [inline Enhancer/ContentRepositoryEnhancer.php]: cs: the closing ) should go on this line, with `) {`

**Comment by dbu** [comment]
i think this makes sense when you have a database layer that does not provide you with relations to automatically load the content. 

can you please check where we can add this to the documentation?

File: Enhancer/ContentRepositoryEnhancer.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Routing\Enhancer;
+
+use Symfony\Cmf\Component\Routing\ContentRepositoryInterface;
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class ContentRepositoryEnhancer implements RouteEnhancerInterface
```
Comment (dbu) [inline Enhancer/ContentRepositoryEnhancer.php]: please add a short description what this enhancer is doing

Comment (andrey1s) [inline Enhancer/ContentRepositoryEnhancer.php]: can you add rules for the `styleci.io`?

File: Enhancer/ContentRepositoryEnhancer.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+     */
+    private $target;
+
+    /**
+     * @var string
+     */
+    private $source;
+
+    /**
+     * @param ContentRepositoryInterface $contentRepository repository for the search content
```
Comment (WouterJ) [inline Enhancer/ContentRepositoryEnhancer.php]: Maybe "repository to search for the content" is better?

File: Enhancer/ContentRepositoryEnhancer.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+     * @param string                     $target            the field name of the content id
+     * @param string                     $source            the field name to set from the map
```
Comment (WouterJ) [inline Enhancer/ContentRepositoryEnhancer.php]: seems like these 2 descriptions should be swapped (`$target` is the new default value and `$source` is the name of the content id).

File: Enhancer/ContentRepositoryEnhancer.php [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        $target = RouteObjectInterface::CONTENT_OBJECT,
+        $source = RouteObjectInterface::CONTENT_ID
+    ) {
+        $this->contentRepository = $contentRepository;
+        $this->target = $target;
+        $this->source = $source;
+    }
+
+    /**
+     *{@inheritdoc}
```
Comment (WouterJ) [inline Enhancer/ContentRepositoryEnhancer.php]: missing space before `{@inheritdoc}`

**Comment by WouterJ** [comment]
I like this feature, it's a very effective and simple way to support content in ORM databases.

**Comment by dbu** [comment]
squashed and merged in 8f172869f59e2fe70ec5b2525c2807b665b18d6c

thanks a lot @andrey1s

**Comment by dbu** [comment]
i created https://github.com/symfony-cmf/RoutingBundle/issues/337 for the bundle. @andrey1s would you have time to work on this?

**Pull request closed without merging**